ontopy/ontology.py

```diff
@@ -1089,3 +1104,31 @@ def new_entity(self, name, parent):
         with self:
             e = types.new_class(name, (parent, ))
         return e
+
+
+class BNode:
+    """Represents a blank node."""
+    def __init__(self,
+                 onto: Union[World, Ontology],
+                 storid: int):
+        """Initiate a blank node.
+
+        Args:
+            onto: Ontology or World instance.
+            storid: The storage id of the blank node.
+        """
+        if (storid >= 0):
+            raise ValueError(
+                f'bnode is supposed to have a negative storid: {storid}')
+        self.onto = onto
+        self.storid = storid
+
+    def __repr__(self):
+        return repr(f'_:b{-self.storid}')
+
+    def __hash__(self):
+        return hash((self.onto, self.storid))
```

Comment:
```suggestion
        return hash(tuple(getattr(self, _) for _ in sorted(self.__dict__))
```
To make sure all attributes (and future attributes) are always given to the hash-function in an alphabetically sorted tuple of attribute values.